accepted/00083-image_management.md

```diff
@@ -0,0 +1,311 @@
+
+   Change foreign key of `suseImageRepoDigest` table from `suseImageBuildHistory.id` to `suseImageInfo.id`.
+
+   For Docker the store can hold only one `name-version` image. Pushing new revision
+   to the store replaces the old one. So the old `suseImageInfo` entries can be set to `obsolete` when the
+   new image is built successfully.
+   On failed build the store still contains the previous revision and the corresponding entry
+   in `suseImageInfo` stays unchanged.
+
+   For details see the Image building workflow below.
+
+   For Kiwi it is possible to have multiple revisions of the same `name-version`. So the entries
+   can be deleted individually. Typically an old image can be deleted when the new image is
+   synced to all Branch servers.
+
+   To maintain previous functionality, add functions to 'delete all revisions'
+   and 'delete all obsolete revisions' of an image.
+
+   `suseImageBuildHistory` will need special attention during migration - it will be necessary
+   to create new row in `suseImageInfo` for each row from `suseImageBuildHistory`
+
+4. Add `pillar_id` to `suseImageInfo`.
+
+   Add table `suseImageFiles` with columns
+   - `id`
+   - `image_id`
+   - `file`
+   - `type`
+   - `external`
+
+   `pillar_id` points to pillar ID in `suseSaltPillars` table.
+
+   Note: According to the RFC https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00080-jsonb-pillar-db-storage.md
+   there is a constraint `UNIQUE (org_id, category)` in `suseSaltPillars`.
+   The `category` for image pillars will be set to `"image_" + suseImageInfo.id`
+
+   `suseImageFiles` contains file or files related to an image.
+   At the beginning there will be only one file of type `bundle`
+   allowed. Later we can extend it to support multiple files per image
+   with types `kernel`, `initrd`, etc.
+
+   `file` is a local path for images in SUMA Store or URL for
+   external images (determined by `external` flag).
+
+   Pillars and local image files will be deleted when the image is deleted
+   in UI or API.
+
+   `file` will be used to show image URL in UI.
+
+   Pillar entries will be deleted when the corresponding `suseImageInfo` entry is deleted.
+   This will be implemented with an sql trigger.
+
+5. Add `suseDeltaImageInfo` table with columns
+   - `sourceImageID`
+   - `targetImageID`
+   - `pillar_id`
+   - `file`
+
+   Add API to create/delete entries, use it in `retail_create_delta`.
+
+   Display deltas in Image details UI.
+
+   Deleting source or target image would delete also the delta image.
+
+6. Store build logs in the database
+
+   - add `log` TEXT column to `suseImageInfo` table
+   - collect the logs at the end of build action, use Salt `cp.push` module
+   - add UI to display it
+
+
+7. External stores
+
+   Make sure that API allows moving the image from SUMA to external store,
+   with adjusting pillars etc.
+
+   Implement a tool that moves the image from SUMA to external server or CDN.
+
+8. API
```

Comment:
Limited to API you should we have UI for it also?

Reply:
I think that these actions mainly make sense as part of more complex workflow, which should be scripted.